internal/controller/quota_tracker.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2023 IBM Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"strings"
+
+	mcadv1beta1 "github.com/project-codeflare/mcad/api/v1beta1"
+	v1 "k8s.io/api/core/v1"
+)
+
+const DefaultResourceLimitsPrefix = "limits."
+
+// QuotaTracker : A tracker of allocated quota, mapped by namespace
+type QuotaTracker struct {
+	state map[string]*QuotaState
+}
+
+// NewQuotaTracker : Create a new QuotaTracker
```

Comment:
```suggestion
// NewQuotaTracker creates a new QuotaTracker
```

AFAIK, the convention in Go is that a doc comment of a func starts with a complete sentence that begins with the name of the element it describes.
ref: https://go.dev/blog/godoc
Although, it seems that this convention is not consistently used in this repo...